tests/lib/rules/no-common-typos.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * @fileoverview Warn for common attribute typos.
+ * @author Koen Punt
+ */
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+var rule = require('../../../lib/rules/no-common-typos');
+var RuleTester = require('eslint').RuleTester;
+
+require('babel-eslint');
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+var options = [
+  {
+    pattern: /dangerouslySetInnerHTML/i,
+    correct: 'dangerouslySetInnerHTML'
+  }
+];
+
+var ruleTester = new RuleTester();
+ruleTester.run('no-common-typos', rule, {
+
+  valid: [{
+    code: [
+      'var Hello = React.createClass({',
+      '  render: function() {',
+      '    return <div dangerouslySetInnerHTML={{__html: "Hello World"}}/>;',
+      '  }',
+      '});'
+    ].join('\n'),
+    options: options,
+    parser: 'babel-eslint',
```

Comment:
The default parser, but when `ecmaVersion` set to 6 or higher, and `jsx` set to true.

It should use the default parser because that's the important one it works with - if you also want to test that it works with babel-eslint that's fine, but that's secondary.